crates/transaction-emitter-lib/src/transaction_generator/call_custom_modules.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    publishing::{module_simple::EntryPoints, publish_util::Package},
+    TransactionExecutor,
+};
+use crate::transaction_generator::{
+    publishing::publish_util::PackageHandler, TransactionGenerator, TransactionGeneratorCreator,
+};
+use aptos_infallible::RwLock;
+use aptos_logger::{info, sample, sample::SampleRate, warn};
+use aptos_sdk::{
+    transaction_builder::TransactionFactory,
+    types::{transaction::SignedTransaction, LocalAccount},
+};
+use async_trait::async_trait;
+use rand::{prelude::StdRng, seq::SliceRandom, Rng};
+use rand_core::{OsRng, SeedableRng};
+use std::{sync::Arc, time::Duration};
+
+pub struct CallCustomModulesGenerator {
+    rng: StdRng,
+    txn_factory: TransactionFactory,
+    packages: Arc<Vec<Package>>,
+    accounts_pool: Option<Arc<RwLock<Vec<LocalAccount>>>>,
+    entry_point: EntryPoints,
+}
+
+impl CallCustomModulesGenerator {
+    pub fn new(
+        rng: StdRng,
+        txn_factory: TransactionFactory,
+        packages: Arc<Vec<Package>>,
+        accounts_pool: Option<Arc<RwLock<Vec<LocalAccount>>>>,
+        entry_point: EntryPoints,
+    ) -> Self {
+        Self {
+            rng,
+            txn_factory,
+            packages,
+            accounts_pool,
+            entry_point,
+        }
+    }
+}
+
+#[async_trait]
+impl TransactionGenerator for CallCustomModulesGenerator {
+    fn generate_transactions(
```

Comment:
nit: this function could use some documentation around the behavior - particularly around burning existing accounts.

Reply:
I am going to do that in a #6187, where I extract burning into a reusable wrapper that can be applied to any txn emitter